llvm/lib/CodeGen/SelectionDAG/SelectionDAG.cpp

```diff
@@ -9708,6 +9710,16 @@ SDValue SelectionDAG::getNode(unsigned Opcode, const SDLoc &DL, EVT VT,
   return getNode(Opcode, DL, VT, Ops, Flags);
 }
 
+SDValue SelectionDAG::getNode(unsigned Opcode, const SDLoc &DL, EVT VT,
```

Comment:
The logic seems different between this function and `getNegative`. `getNegative` will use the VP version `SUB` automatically if we have a non-null `VPOp`. But these 2 new `getNode` ask the user to chose right opcode even giving a non-null `VPOp`. It's a little bit inconsistent.